src/Arcus.EventGrid.Security.Core/Validation/EventGridSubscriptionValidator.cs

```diff
@@ -69,44 +67,41 @@ public async Task<IActionResult> ValidateEventGridSubscriptionEventRequestAsync(
         {
             Guard.NotNull(request, nameof(request), "Requires a HTTP request to validate the Azure EventGrid subscription event");
 
-            string json = await ReadRequestBodyAsync(request);
-            if (string.IsNullOrWhiteSpace(json))
+            BinaryData data = await BinaryData.FromStreamAsync(request.Body);
```

Comment:
TIL.  I wasn't aware of this BinaryData type.  Seems like this is just a wrapper around a byte array ?